src/lib/backend/equinoxe_cohttp.ml

```diff
@@ -34,56 +32,31 @@ module Backend = struct
   let bind f m = Lwt.bind m f
   let fail (`Msg e) = Lwt.fail_with e
 
-  (***** Helpers *****)
+  (***** Helper *****)
 
-  let compute ~time ~f =
-    let* status =
-      Lwt.pick
-        [
-          (f () >|= fun v -> `Done v);
-          (Lwt_unix.sleep time >|= fun () -> `Timeout);
-        ]
-    in
-    match status with
-    | `Done v -> Lwt.return v
-    | `Timeout -> Lwt.fail_with "Http request timeout"
-
-  let get_body (resp, body) =
-    let code = Cohttp.(Response.status resp |> Code.code_of_status) in
-    if code >= 200 && code < 300 then Cohttp_lwt.Body.to_string body
-    else
-      let msg = Format.sprintf "Cohttp exits with HTTP code %d" code in
-      Lwt.fail_with msg
+  let compute fn ~headers ~url =
+    let headers = Cohttp.Header.of_list headers in
+    let url = Uri.of_string url in
+    Lwt.bind (fn ~headers ~url) (fun (_, body) ->
```

Comment:
Just a question, otherwise it looks great to me: if we don't raise an error anymore when the code is wrong (not in [200 ; 300]), wouldn't the user only get a JSON error instead of one coming from the HTTP client?

Reply:
The HTTP client can still raise lwt errors (if the http protocol goes wrong for example)... but yes we loose the "http status code" that could indicate an error. The motivation was that the error returned by Equinix in the json:

https://github.com/maiste/equinoxe/blob/b557072f8486ead082d5c763f7064218364d9da1/src/lib/equinoxe/equinoxe.ml#L57-L61

is a lot more descriptive than `"Cohttp exits with HTTP code 400"` (vs `"Invalid token"`). There is a grey area if we get an http error code but no error in the json from Equinix: but I believe we should still use the json in that case?

Reply:
Yes in my tests it always returns a nice JSON error! But I will keep an eye on it just in case we really do need to catch those HTTP errors :)

Reply:
Sure, we can still add it later if we encounter a specific case :+1: